### PR TITLE
Hero selection

### PR DESCRIPTION
**Pull request by filwas**
I've encapsulated the topmost text + progress bar. 

Chosen hero's index is now persisted via html attribute. Choose-fighter-step.js line 9 and Wizard.js line 35. I'm not really sure if this is a good solution, but I have no idea how to implement it otherwise.

### CODE REVIEW

File: js/wizard/steps/choose-fighter-step/characterProfileMaker.js [hunk 1]
```diff
@@ -1,8 +1,5 @@
-import { characterToggler } from './characterToggler.js';
-
-export function characterProfileMaker(index) {
+export function characterProfileMaker(characterDetails) {
```
Comment (Sumick) [inline js/wizard/steps/choose-fighter-step/characterProfileMaker.js]: **nitpick:** If it's a maker function I think you should keep it together with other `makers` in `component` dir

File: js/screens/wizard-screen.js [hunk 2]
```diff
@@ -2,6 +2,7 @@ import { initWizard } from '../wizard/wizard.js';
 
 export const wizardScreen = () => {
    const wizardScreenWrapper = document.createElement('div');
```
Comment (Sumick) [inline js/screens/wizard-screen.js:5]: **suggestion:** I think this will be a good place to keep some state for your wizard, so it wraps all steps

**Review by Sumick — CHANGES_REQUESTED** [review]
Looks good, but I don't see any logic for persisting chosen Hero in the wizard state. In this implementation when you will change wizard step from hero selection to booking details, you will loose you selected hero data.

**Comment by filwas** [comment]
I couldn't find any other easy way to store the data persistently other than to set a custom attribute for wizardScreenWrapper: 

1. WizardScreenWrapper is initialised with characterIndex = 0
2. Now, the wizard itself requires a props object, which contains wizard's step and the index of chosen character. 
3. In choose-fighter-step, the wizardScreenWrapper characterIndex attribute is updated with every arrow button click.
4. Whenever user clicks the wizard buttons, a fresh wizard is created. We're reading the last chosen character from wizardScreenWrapper's attribute, and pass it on into the props object of the new wizard.

Maybe a better solution would be creating a fresh wizard state on every arrow button click? That way all the information would be passed around in props only.

Or better yet, let's just put this record on loop instead as the wizard: https://www.youtube.com/watch?v=xsbAVGDPvM0

**Comment by filwas** [comment]
> Maybe a better solution would be creating a fresh wizard state on every arrow button click? That way all the information would be passed around in props only.

Yeah, I did just that. :)

File: js/components/characterData.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
+export const characterData = [
```
Comment (Sumick) [inline js/components/characterData.js]: **nitpick:** This is not a component but a dumb data - it should not be placed here

File: js/screens/wizard-screen.js [hunk 4]
```diff
@@ -2,7 +2,11 @@ import { initWizard } from '../wizard/wizard.js';
...
-   const wizard = initWizard(0);
+   wizardScreenWrapper.id = 'wizardScreenWrapper';
+   const wizard = initWizard({
+      wizardStep: 0,
+      characterIndex: 0,
```
Comment (Sumick) [inline js/screens/wizard-screen.js]: **issue:** Keeping selected character index is not a good idea, as the data order can change in time. I suggest you to assign **whole selected fighter object**, or fighter `id` - but then you'd have to add `id` to your data. Then, when you select another fighter, you just assign other object value. Example:

```js
const wizard = initWizard({
      wizardStep: 0,
      // should be null when nothing is selected
      selectedCharacter: null,
```

**Review by Sumick — CHANGES_REQUESTED** [review]
Looks good, I found one thing to improve - the way we should keep selected fighter in state.

File: js/components/characterData.js (hunk 3 above)
Comment (filwas) [reply js/components/characterData.js]: Could you elaborate more? How else should I store the character data?

EDIT: NVM, I misunderstood what you meant. I'll make a "dumb data" folder now. :)

File: js/screens/wizard-screen.js (hunk 4 above)
Comment (filwas) [reply js/screens/wizard-screen.js:8]: This seems simple enough, but I'm not sure about accessing the data itself. Previously you said it's not a problem to have all of the character data stored as a file to be exported and shared around the functions as na argument, did I implement it wrong? When I tried to have it as a .JSON file, I was receiving some errors when trying to access it.

**Review by Sumick — APPROVED** [review]
Looks good!

**Pull request merged**